be/src/common/statusor.h

```diff
@@ -702,4 +703,29 @@ void StatusOr<T>::IgnoreError() const {
     // no-op
 }
 
+#define ASSIGN_OR_RETURN_IMPL(varname, lhs, rhs) \
+    auto&& varname = (rhs);                      \
```

Comment:
use a `do {
}while(0)` to enclose this macro.

Reply:
Enclosing the macro inside a `do {} while(0)` will disable extracting the value into a new variable, like this
```
ARROW_ASSIGN_OR_RETURN(auto value, MaybeGetValue(arg));
```

Reply:
`varname` is generated by the macro `VARNAME_LINENUM`, which will add the file number as the suffix of `varname`, no naming conflict in practice